linkerd/app/outbound/src/discover.rs

```diff
@@ -68,109 +72,186 @@ pub(crate) fn resolver(
-                );
-                tracing::debug!(
-                    ?policy,
-                    "Policy not found; no profile: synthesizing policy from original destination address"
                 );
-
-                let (tx, policy_rx) = watch::channel(policy);
-                // keep the watch channel alive as long as it's in use.
-                tokio::spawn(async move { tx.closed().await });
-                return Ok((None, policy_rx));
+                return Ok((Some(profile), policy));
             }
 
-            // Otherwise, errors resolving the policy are considered fatal.
-            Err(policy_error)
+            // Otherwise, route the request to the original destination address.
+            let policy = spawn_synthesized_origdst_policy(orig_dst, queue, detect_timeout);
+            Ok((None, policy))
         })
     })
 }
 
+#[inline]
+fn is_not_found(e: &Error) -> bool {
+    e.downcast_ref::<tonic::Status>()
+        .map(|s| s.code() == tonic::Code::NotFound)
+        .unwrap_or(false)
+}
+
+fn spawn_synthesized_profile_policy(
+    orig_dst: SocketAddr,
+    mut profile: watch::Receiver<profiles::Profile>,
+    queue: policy::Queue,
+    detect_timeout: Duration,
+) -> watch::Receiver<policy::ClientPolicy> {
+    let mk = move |profile: &profiles::Profile| {
+        let (addr, meta) = profile
+            .endpoint
+            .clone()
+            .unwrap_or_else(|| (orig_dst, Default::default()));
+        // TODO(ver) We should be able to figure outc resource coordinates for
+        // the endpoint?
+        synthesize_forward_policy("endpoint", detect_timeout, queue, addr, meta)
+    };
+
+    let policy = mk(&*profile.borrow_and_update());
+    tracing::debug!(?policy, profile = ?*profile.borrow(), "Synthesizing policy from profile");
+    let (tx, rx) = watch::channel(policy);
+    tokio::spawn(async move {
+        let mut profile = profile;
+        loop {
+            if profile.changed().await.is_err() {
+                tracing::debug!("Profile watch closed, terminating");
+                return;
+            };
+            let policy = mk(&*profile.borrow());
+            tracing::debug!(?policy, "Profile updated; synthesizing policy");
+            if tx.send(policy).is_err() {
+                tracing::debug!("Policy watch closed, terminating");
+                return;
+            }
+        }
+    });
+    rx
+}
+
+fn spawn_synthesized_origdst_policy(
+    orig_dst: SocketAddr,
+    queue: policy::Queue,
+    detect_timeout: Duration,
+) -> watch::Receiver<policy::ClientPolicy> {
+    let policy = synthesize_forward_policy(
+        "fallback",
+        detect_timeout,
+        queue,
+        orig_dst,
+        Default::default(),
+    );
+    tracing::debug!(?policy, "Synthesizing policy");
+    let (tx, rx) = watch::channel(policy);
+    tokio::spawn(async move {
+        tx.closed().await;
+    });
+    rx
+}
+
+fn synthesize_forward_policy(
+    name: impl ToString,
```

Comment:
nit, TIOLI: i feel like this could maybe take an `Arc<Meta>` rather than an `impl ToString` --- we use hard-coded strings ("endpoint" and "fallback") for the name, so that could be a singleton rather than creating a new `Arc<Meta>` for every synthesized policy? not a blocker.

Reply:
keep in mind this is going to happen at most once per connection (and less frequently than that in practice).